src/main/java/uk/gov/hmcts/reform/professionalapi/controller/internal/OrganisationInternalController.java

```diff
@@ -127,6 +129,55 @@ public ResponseEntity retrievePaymentAccountBySuperUserEmail(@NotNull @RequestPa
         return retrievePaymentAccountByUserEmail(email);
     }
 
+    @ApiOperation(
+            value = "Edit the PBAs of an Organisation by Organisation ID",
+            authorizations = {
+                    @Authorization(value = "ServiceAuthorization"),
+                    @Authorization(value = "Authorization")
+            }
+    )
+    @ApiResponses({
+            @ApiResponse(
+                    code = 200,
+                    message = "The Organisation's associated payment accounts",
+                    response = PbaResponse.class
+            ),
+            @ApiResponse(
+                    code = 403,
+                    message = "Forbidden Error: Access denied"
+            ),
+            @ApiResponse(
+                    code = 404,
+                    message = "Data not found"
+            )
+    })
+    @PutMapping(
+            path = "/{orgId}/pbas",
+            produces = MediaType.APPLICATION_JSON_UTF8_VALUE
+    )
+    @Secured("prd-admin")
+    public ResponseEntity editPaymentAccountsByOrgId(@Valid @NotNull @RequestBody PbaEditRequest pbaEditRequest,
+                                                     @PathVariable("orgId") @NotBlank String organisationIdentifier) {
+        log.info("Received request to edit payment accounts by organisation Id...");
+
+        Organisation organisation = organisationService.getOrganisationByOrgIdentifier(organisationIdentifier);

```

Comment:
The signature of this service should be changed to return an optional

Reply:
Changing the return type of this service affected a numerous amounts of unit and integration tests and other services. Instead I have wrapped the result in an Optional.